R/model-wrappers.R

```diff
@@ -290,3 +290,173 @@ forecastHybrid_model <- function(y = NULL, samples = NULL,
+#' `function(...){EpiSoon::bsts_model(model = , ...)}`.
+#' @inheritParams bsts_model
+#' @param weighting_period The number of most recent timepoints to hold out to
+#' generate the weights for the mixture model
+#' @return A dataframe of predictions (with columns representing the
+#'  time horizon and rows representing samples).
+#' @export
+#' @importFrom purrr map_dfr
+#' @importFrom dplyr bind_rows n ungroup mutate rename group_by filter
+#' @importFrom tidyr pivot_longer pivot_wider
+#' @examples
+#'
+#' \dontrun{
+#'
+#' # make list with models
+#' models <- list(
+#'   "ARIMA" = function(...){EpiSoon::fable_model(model = fable::ARIMA(y), ...)},
+#'   "ETS" = function(...){EpiSoon::fable_model(model = fable::ETS(y), ...)},
+#'   "Drift" = function(...){EpiSoon::fable_model(model = fable::RW(y ~ drift()), ...)}
+#' )
+#'
+#' # make forecst on its own
+#' forecast <- stackr_mixture_model(y = EpiSoon::example_obs_rts[1:10, ]$rt
+#'                                  models = models,
+#'                                  samples = 10,
+#'                                  horizon = 7,
+#'                                  weighting_period = 5)
+#'
+#'
+#' # together with forecast_rt
+#' fc_rt <- forecast_rt(EpiSoon::example_obs_rts[1:10, ],
+#'                      model = function(...){
+#'                        stackr_mixture_model(models = models,
+#'                                             weighting_period = 5,
+#'                                             ...)},
+#'                      samples = 10,
+#'                      horizon = 7)
+#'
+#' forecast_eval <- evaluate_model(EpiSoon::example_obs_rts,
+#'                                 EpiSoon::example_obs_cases,
+#'                                 model = function(...){
+#'                                   stackr_mixture_model(models = models,
+#'                                                        weighting_period = 5,
+#'                                                        ...)},
+#'                                 horizon = 7, samples = 10,
+#'                                 serial_interval = example_serial_interval,
+#'                                min_points = 10)
+#'
+#' plot_forecast_evaluation(forecast_eval$forecast_rts,
+#'                          EpiSoon::example_obs_rts,
+#'                          horizon_to_plot = 7)
+#'
+#' }
+#'
+
+stackr_mixture_model <- function(y = NULL,
+                                 models = NULL,
+                                 samples = NULL,
+                                 horizon = NULL,
+                                 weighting_period = 5) {
+
+
+  #### Error Handling
+  # check if y is there
+  if(is.null(y)) stop("parameter y is missing")
+
+
+  # check if stackr is installed
+  check_suggests("stackr",
+                 dev_message = "Install using devtools::install_github('nikosbosse/stackr')")
+
+  if (length(y) <= weighting_period) {
+    stop("not enough observations to do weighting. Adjust weighting_period")
```

Comment:
Capital

Reply:
Perhaps it should just not do a mixture model if not enough data and instead, return a mean ensemble. This set up will be very annoying when running across many regions